app/models/odk/response_parser.rb

```diff
@@ -178,10 +178,8 @@ def lookup_and_check_form(params)
       raise "xml submissions must be to versioned forms" if form.current_version.nil?
 
       # if form version is outdated, error
-      if form.current_version.code != params[:version] &&
-          form.current_version.number != params[:version]
-        raise FormVersionError, "Form version is outdated"
-      end
+      oldest_accepted_num = form.oldest_accepted_version.number.to_i
+      raise FormVersionError, "Form version is outdated" if params[:version].to_i < oldest_accepted_num
```

Comment:
We still need to handle three letter codes for now b/c there will be forms out there with them. Perhaps the logic is:

* If it looks like a three letter code, it has to match one of the accepted versions' code exactly (you can put a comment on this that it can be removed once we're confident those are not in the wild anymore)
* Otherwise do the `<` comparison.